pkg/container-collection/containers.go

```diff
@@ -118,14 +118,17 @@ type ContainerSelector struct {
 // to help users to identify the workflow of the profile. We "lazily
 // enrich" this information because this operation is expensive and this
 // information is only needed in some cases.
-func (c *Container) GetOwnerReference() (*metav1.OwnerReference, error) {
+func (c *Container) GetOwnerReference(kubeconfig *rest.Config) (*metav1.OwnerReference, error) {
 	if c.K8s.ownerReference != nil {
 		return c.K8s.ownerReference, nil
 	}
 
-	kubeconfig, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, fmt.Errorf("getting Kubernetes config: %w", err)
+	if kubeconfig == nil {
+		var err error
+		kubeconfig, err = rest.InClusterConfig()
```

Comment:
This logic of using the specified kubeconfig and falling back on InClusterConfig is not applied everywhere:
```bash
$ git grep InClusterConfig -- pkg/container-collection/
pkg/container-collection/containers.go:126:     kubeconfig, err := rest.InClusterConfig()
pkg/container-collection/k8s.go:51:     config, err := rest.InClusterConfig()
pkg/container-collection/options.go:508:                        kubeconfig, err = rest.InClusterConfig()
pkg/container-collection/podinformer.go:53:     config, err := rest.InClusterConfig()
```

and:

```bash
$ git grep NewClientset -- pkg/operators/
pkg/operators/common/kubeinventorycache.go:79:  clientset, err := k8sutil.NewClientset("")
pkg/operators/oci-handler/oci.go:130:   k8sClient, err := k8sutil.NewClientset("")
```

So the patch about adding kubeconfig is incomplete.

I would like to first get the support for OwnerReference enrichment when the `ig` executable is used in Kubernetes pods with in-cluster communications.

Could you move the kubeconfig patch in a separate PR, so we can get the basic OwnerReference enrichment support earlier?

Reply:
Split into #3370